### ISSUE TITLE
Load failure, gene missing: SPBC8E4.02c

### ISSUE THREAD

**Issue by kimrutherford** [pombase/curation#2210]
Hi.

The load failed last night because there is now no gene with the systematic ID "SPBC8E4.02c".  There are Canto sessions that refer to the gene and unfortunately the code that reads the Canto data into Chado is no very forgiving about missing genes.  There were warning from the BioGRID and phenotype loading code as well but those didn't cause the load to stop.

This is the change to chromosome2.contig that caused the problem:

-FT                   /systematic_id="SPBC8E4.02c"
-FT                   /controlled_curation="term=species distribution,
-FT                   Schizosaccharomyces pombe specific; date=20140820"
+FT                   /systematic_id="spncrna.9001"
+FT                   /synonym="SPBC8E4.02c"

Is it possible to swap the values of the /synonym and the /systematic_id for this feature?

**Comment by Antonialock** [comment]
@ValWood this is ncRNA prt2, I gave it a new sysID (ncrna.9001) 
what should we do here?

**Comment by ValWood** [comment]
We need to find the canto session that refer to this gene and fix the name here. Although this is odd, I would expect these to update automatically...I'm not sure to be honest. WIll dig around...

**Comment by ValWood** [comment]
I think we will need to migrate all existing annotations to
ncrna.9001
(the system will not know that the ID has been decommissioned and  a new ID enstated)

This will involvve updating the Marguerat  expression files
Informing BioGRID
Genes which have SPBC8E4.02c as an extension
phenotypes
GO

(@kimrutherford  can any of this be done automagically ?)

**Comment by kimrutherford** [comment]
>  can any of this be done automagically ?

It's going to involve editing any input file containing the ID.  Some of that can be done with Perl.  I'll need to update the Canto database manually.

> Although this is odd, I would expect these to update automatically

The systematic IDs are the primary keys that tie everything together.  They are how the loading code know which gene is which.  If you change a systematic ID it's like adding a gene and removing an gene.  There's no way for the code to know that it was renamed.

**Comment by kimrutherford** [comment]
We might need to get a new PR term too.

This is from mini-ontologies/pombe_mini_PR.obo:

```
[Term]
id: PR:O42886
name: uncharacterized protein C8E4.02c (Schizosaccharomyces pombe 972h-)
def: "A protein that is a translation product of the Schizosaccharomyces pombe 972h- SPBC8E4.02c gene." [PRO:DNx, UniProtKB:O42886]
comment: Category=organism-gene.
synonym: "Spom972h-SPBC8E4.02c" EXACT PRO-short-label [PRO:DNx]
xref: UniProtKB:O42886
is_a: PR:000029043 ! Schizosaccharomyces pombe 972h- protein
```

Although I can't see anywhere the term is used.

**Comment by kimrutherford** [comment]
These file look like they need updating in pombe-embl SVN.  I can sort them out except for `mini-ontologies/pombe_mini_PR.obo` but I don't think that one is urgent.

```
external_data/phaf_files/chado_load/PMID_20473289_phaf.tsv
external_data/phaf_files/chado_load/PMID_25452419_phaf.tsv
external_data/phaf_files/chado_load/PMID_27887640_phaf.tsv
external_data/Quantitative_gene_expression_data/Marguerat_RNA_proliferation.txt
external_data/Quantitative_gene_expression_data/Marguerat_RNA_quiescence.txt
mini-ontologies/pombe_mini_PR.obo
orthologs/pombe_cerevisiae_ortholog_table.txt
phenotype_mapping/phaf_format_phenotypes.tsv
```

There was only one session that had the gene added (3c66d095896b8ee5).  I've fixed that.

> Genes which have SPBC8E4.02c as an extension

I can only see two sessions that do that.  Those occurences should appear in the logs tomorrow as warnings.  The sessions are:

9a6cbb66c43cb971
a93df4140a4b1e4c

**Comment by kimrutherford** [comment]
I missed a file:
  ftp_site/pombe/names_and_identifiers/PomBase2UniProt.tsv

Does the entry need to be removed from UniProt?

There probably others in the ftp_site directory that need fixing but I'll wait until after the next load because quite a few are automatically updated now.

**Comment by kimrutherford** [comment]
I fixed the gene in session 3c66d095896b8ee5 before the nightly load got as far as the Canto data.  So the load finished but spncrna.9001/SPBC8E4.02c is missing the phenotype, gene expression and ortholog data.  That stuff should be back on Sunday morning.

The dev site has the update:
http://pombase-dev.bioinformatics.nz/gene/spncrna.9001

Unfortunately the Ensembl inline image is out of date now.

**Comment by ValWood** [comment]
> Unfortunately the Ensembl inline image is out of date now.

yep quite a few are. we have changes going back to before the new website

**Comment by kimrutherford** [comment]
The load worked and the gene page is OK: https://www.pombase.org/gene/spncrna.9001
except that most of the links in the External references section don't work.

**Comment by mah11** [comment]
I can delete the PR:O42886 entry from pombe_mini_PR.obo; it was only included because I pulled in everything that has "is_a: PR:000029043 ! Schizosaccharomyces pombe 972h- protein" from PRO (not because we're using that particular PRO term in any annotations or extensions).

Getting it out of PRO itself is another matter ... I don't know if they automagically update to reflect changes in protein-coding genes for any species ...

**Comment by ValWood** [comment]
@Antonialock  Is there anything else still to do here? We should get this one closed...
